src/download.rs

```diff
@@ -60,6 +60,9 @@ pub enum DownloadState {
     /// Undecipherable message.
     Undecipherable = 30,
 
+    /// Locally deleted message, should be deleted on the server also.
+    ToDelete = 40,
```

Comment:
No, i don't want it to be an API change, i think we should map it to `Done`, anyway the message can't be displayed after a user-deletion.

> when we receive multi-part message only the message part that is deleted will get `ToDelete` state while other parts with the same Message-ID have different state.

This should be fixed. Now `rfc724_mid_exists_ex()` looks at a random message having the given Message-ID, it's just a bug. We mustn't delete from the server a message having only some parts user-deleted.

> It seems this "ToDelete" state should be associated with rfc724_mid (Message-ID) rather than msg_id (message bubble).

Probably yes, but fixing `rfc724_mid_exists_ex()` is also a way to get this working and it looks simpler than adding a new table. I'd try to fix the current code first, let's see if it works

Reply:
Added a new `msgs.deleted` column instead. It seems easier to tweak `rfc724_mid_exists_ex()` so that it checks all message parts than maintaining the `(timestamp, rfc724_mid)` table.